src/Security/MemberAuthenticator/LostPasswordHandler.php

```diff
@@ -197,16 +189,55 @@ public function forgotPassword($data, $form)
 
         // Avoid information disclosure by displaying the same status,
         // regardless wether the email address actually exists
+        $identifier = isset($data[$uniqueIdentifier]) ? $data[$uniqueIdentifier] : '';
         $link = Controller::join_links(
             $this->link('passwordsent'),
-            rawurlencode($data['Email']),
+            rawurlencode($identifier),
             '/'
         );
 
         return $this->redirect($this->addBackURLParam($link));
     }
 
     /**
+     * Ensure that the user has provided an email address. Note that the "Email" key is specific to this
+     * implementation, but child classes can override this method to use another unique identifier field
+     * for validation.
+     *
+     * @param  array $data
+     * @param  LostPasswordForm $form
+     * @return HTTPResponse|null
+     */
+    protected function validateForgotPasswordData(array $data, LostPasswordForm $form)
+    {
+        if (empty($data['Email'])) {
```

Comment:
We have hard-coded Email here but soft-coded it to unique_identifier_field elsewhere.

Reply:
Also, LostPasswordForm.php has it hard-coded to `Email` as well.

Reply:
If that's the case should you remove the soft-coding on the other part of the class and fix to `Email`?

Reply:
Yep, sure. I think it's a confusion of trying to be abstract and concrete at the same time. I'll change `getMemberFromData` back to the original logic it had.